aten/src/ATen/native/cuda/UpSample.cuh

```diff
@@ -255,5 +255,101 @@ __device__ __forceinline__ static accscalar_t cubic_interp1d(
   return x0 * coeffs[0] + x1 * coeffs[1] + x2 * coeffs[2] + x3 * coeffs[3];
 }
 
+namespace upsample_antialias {
+
+// taken from
+// https://github.com/python-pillow/Pillow/blob/6812205f18ca4ef54372e87e1a13ce4a859434df/
+// src/libImaging/Resample.c#L20-L29
+template <typename accscalar_t>
+__device__ __forceinline__ static accscalar_t bilinear_filter(accscalar_t x) {
+  if (x < 0.0) {
```

Comment:
you probably should cast literals to accscalar_t, I'm not sure they won't be treated as double for float x, and that's a perf penalty